File: AD-HOC/3059/main.py
def total_pairs(n, i, f, vector):
    total = 0
    for p in range(n-1):
        for q in range(p + 1, n):
            pair = vector[p] + vector[q]
            total += i <= pair <= f
    return total


n, i, f = map(int, input().split())
vector = list(map(int, input().split()))
print(total_pairs(n, i, f, vector))

File: AD-HOC/3059/main.rb
# frozen_string_literal:true

def total_pairs(n, i, f, vector)
  total = 0
  (0...n - 1).each do |p|
    (p + 1...n).each do |q|
      pair = vector[p] + vector[q]
      total += 1 if i <= pair && pair <= f
    end
  end
  total
end

n, i, f = gets.split.map(&:to_i)
vector = gets.split.map(&:to_i)
puts total_pairs(n, i, f, vector)
